controllers/datasciencecluster/datasciencecluster_controller.go

```diff
@@ -282,6 +316,7 @@ func (r *DataScienceClusterReconciler) SetupWithManager(mgr ctrl.Manager) error
 		Owns(&appsv1.Deployment{}).
 		Owns(&appsv1.ReplicaSet{}).
 		Owns(&corev1.Pod{}).
+		Watches(&source.Kind{Type: &dsci.DSCInitialization{}}, handler.EnqueueRequestsFromMapFunc(r.watchDataScienceClusterResources)).
```

Comment:
not sure i understand this part: if it related to the subscription config part, or it is to fix the missing piece when DSCI is changed (e.g monitoring from disable to enable) which should trigger reconcile of DSC

looks like, events from DSCI will be watched and trigger watchDataScienceClusterResoures() to return the DSC instance (if it exists) then reconcile it?

Reply:
So this in general adds watch for DSCI instance. It is required for this change because `DSC` will go in error phase if created without DSCInitialization. When the user then creates DSCInitialization, we need to reconcile DSC in order to get out of the error state